modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/io/PageIO.java

```diff
@@ -984,4 +982,22 @@ public static String printPage(long addr, int pageSize) {
 
         return sb.toString();
     }
+
+    /**
+     * Asserts that page type of the page stored at pageAddr matches page type of this PageIO.
+     *
+     * @param pageAddr address of a page to use for assertion
+     */
+    protected final void assertPageType(long pageAddr) {
```

Comment:
I don’t think this method is needed because it is called in hot code and we don’t have 100% guarantee that it will be removed when the asserts are turned off, such a check should only be where it is definitely needed.